src/sentry/integrations/github/client.py

```diff
@@ -43,17 +46,119 @@ def __repr__(self) -> str:
+        expires_at = datetime.strptime(data["expires_at"], "%Y-%m-%dT%H:%M:%SZ").isoformat()
+        integration.metadata.update({"access_token": token, "expires_at": expires_at})
+        integration.save()
+        logger.info(
+            "token.refresh_end",
+            extra={
+                "new_expires_at": integration.metadata.get("expires_at"),
+                "integration_id": integration.id,
+            },
+        )
+
+        self.integration = integration
+        return token
+
+    @control_silo_function
+    def _get_token(self, prepared_request: PreparedRequest) -> str | None:
+        """
+        Get token retrieves the active access token from the integration model.
+        Should the token have expired, a new token will be generated and
+        automatically persisted into the integration.
+        """
+        self.integration: RpcIntegration
+
+        if not self.integration:
+            return None
+
+        logger_extra = {
+            "path_url": prepared_request.path_url,
+            "integration_id": self.integration.id,
+        }
+
+        # Only certain routes are authenticated with JWTs....
+        should_use_jwt = prepared_request.path_url.startswith("/app/installations")
+        if should_use_jwt:
+            token = get_jwt()
+            logger.info("token.jwt", extra=logger_extra)
+            return token
+
+        # The rest should use access tokens...
+        now = datetime.utcnow()
+        token: str | None = self.integration.metadata.get("access_token")
+        expires_at: str | None = self.integration.metadata.get("expires_at")
+        is_expired = bool(expires_at) and datetime.strptime(expires_at, "%Y-%m-%dT%H:%M:%S") < now
+        should_refresh = not token or not expires_at or is_expired
+
+        if should_refresh:
+            token = self._refresh_access_token()
+
+        logger.info("token.access_token", extra=logger_extra)
+        return token
+
+    @control_silo_function
+    def authorize_request(self, prepared_request: PreparedRequest) -> PreparedRequest:
+        integration = None
+        if self.integration:
+            integration = self.integration
+        elif self.integration_id:
+            integration = Integration.objects.filter(
+                id=self.integration_id,
+                provider=EXTERNAL_PROVIDERS[ExternalProviders.GITHUB],
+                status=ObjectStatus.ACTIVE,
+            ).first()
+
+        if not integration:
+            logger.info("no_integration", extra={"path_url": prepared_request.path_url})
+            return prepared_request
+
+        token = self._get_token(prepared_request=prepared_request)
+        if not token:
+            logger.info(
+                "no_token",
+                extra={"path_url": prepared_request.path_url, "integration_id": integration.id},
+            )
+            return prepared_request
+
+        prepared_request.headers["Accept"] = "application/vnd.github+json"
```

Comment:
This is the recommended 'Accept' header from the GitHub API ([example](https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#list-repositories-accessible-to-the-app-installation))

This can likely also remove the legacy comments:
```
# TODO(jess): remove this whenever it's out of preview
"Accept": "application/vnd.github.machine-man-preview+json",
```
but I've only done this change for the core github integration, there are still comments like this for GitHub Enterprise and the plugin.

Reply:
This should be fine. We might get burned by old versions of github-enterprise. These content-types were graduated back in 2020 https://developer.github.com/changes/2020-08-20-graduate-machine-man-and-sailor-v-previews/. I don't know how far back we intend to support github-enterprise but 2 years feels like a reasonable window.